### PR TITLE
Make sure `src/SDKs/_metadata` never exists

### LINKED ISSUE #10924: Prevent generation of src/SDKs/_metadata/ Directory
Track down the actual reason why _metadata is still being generated in the old repo structure, stop this from happening then clean up `src/SDKs/_metadata/` directory.

### PR DESCRIPTION
**Pull request by isra-fel**
## Description
This PR adds a check to management plane PRs. It checks for old metadata location `src/SDKs/_metadata`. It fails if it exists.
~~Currently the script is checked in in this repo. After verified, I'll move it to Azure/azure-sdk-tools repo.~~
The script will be inline in `archetype-sdk-mgmt.yml`.

Resolves #10924

## How to verify
- [x] If there's file in `src/SDKs/_metadata`, the job `net - mgmt - ci (Analyze)` should fail.
  - [x] There should be a message to guide the user to update build tools
  - ADO build: https://dev.azure.com/azure-sdk/public/_build/results?buildId=353798&view=logs
- [x] If there isn't this directory, the job should succeed
  - ADO build: https://dev.azure.com/azure-sdk/public/_build/results?buildId=353838

### CODE REVIEW

File: eng/pipelines/templates/scripts/verify-metadata-location.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# This check fails when there's file in `src/SDKs/_metadata/` directory
```
Comment (weshaggard) [inline eng/pipelines/templates/scripts/verify-metadata-location.yml:1]: I wouldn't do this as a template. I'd just do it inline. the mgmt.yml.

If you do decide to keep the template can you please put it under `eng/pipelines/templates/steps` scripts folder is intended for actual scripts not yml files.

Comment (isra-fel) [reply eng/pipelines/templates/scripts/verify-metadata-location.yml]: Great! Moved the scripts inline.

**Review by chidozieononiwu — APPROVED** [review]

**Review by weshaggard — APPROVED** [review]

**Comment by isra-fel** [comment]
Thank you for reviewing!

**Pull request merged**